### DISCUSSION TITLE
Dynamic field in the document

### DISCUSSION THREAD

**Discussion by mattewid** [BeanieODM/beanie#581] [Question]
**Hello**,

I'm trying to migrate from MongoEngine to Beanie little by little. Now that I'm starting to test Beanie and I'm having some doubts.

Doubts:
**1** - How to get (Document.dynamic_field) without this field being defined in the Document? For I tried and got the error AttributeError.
**2** - Due to my doubt 1, this doubt arose to know if it is also possible to save fields not defined in the Document.

I looked for something in the documentation and I didn't find it.

**Comment by roman-right** [comment]
Hi,
I think you can use extra fields from pydantic for this. Check this doc: https://docs.pydantic.dev/latest/usage/model_config/#options

**Reply by mattewid** [reply]
Thanks for showing me this, looks like it's what I'm looking for.

However, what is the correct way to use this? Where to set? Because I added the Configs class inside the MyModel(Document). However, I believe this is not the case, because when adding this Configs class inside the model, wouldn't I be overwriting the Configs defined by the Beanie in the Document class? This class: [Ref](https://github.com/roman-right/beanie/blob/02111b0ceb673227b51b2233029cc71d8a92d30c/beanie/odm/documents.py#L1087)

Sorry if I'm asking something silly. Because it's my first time using Beanie and Pydantic.

Here is my code as an example:
```python

class MyModel(Document):

    created_in: datetime = Field(default_factory=datetime.utcnow)

    # Pydantic configs: https://docs.pydantic.dev/latest/usage/model_config/#options
    class Config():
        extra = Extra.allow


    # Beanie settings: https://beanie-odm.dev/tutorial/defining-a-document/#settings
    class Settings():
        name = 'mymodel_coll'
        # keep_nulls = False
        indexes = [
            IndexModel(
                [
                    ('status', pymongo.ASCENDING)
                ]
            )
        ]


    
```

**Reply by gsakkis** [reply]
> when adding this Configs class inside the model, wouldn't I be overwriting the Configs defined by the Beanie in the Document class? 

That's exactly what would happen in "regular" Python. Pydantic however uses [metaclass magic](https://github.com/pydantic/pydantic/blob/main/pydantic/v1/main.py#L121) to [inherit stuff](https://github.com/pydantic/pydantic/blob/main/pydantic/v1/main.py#L136-L145) (such as [configs and validators](https://github.com/pydantic/pydantic/blob/main/pydantic/v1/main.py#L157-L159) among others) from base classes instead of overwriting them.

**Reply by mattewid** [reply]
It was really good to know that. Thank you very much.
